### PR TITLE
runtime: Add join function on Array and ArraySlice

### PR DESCRIPTION
**Pull request by robryanx**
This takes advantage of the existing string formatters to allow an Array
or ArraySlice of any type to be joined into a string.

### CODE REVIEW

File: runtime/prelude.jakt [hunk 1]
```diff
@@ -22,6 +22,7 @@ extern struct Array<T> {
     function iterator(this) -> ArrayIterator<T>
     function first(this) -> T?
     function last(this) -> T?
+    function join(this, anon s: String) throws -> String
```
Comment (networkException) [inline runtime/prelude.jakt:25]:
```suggestion
    function join(this, anon separator: String) throws -> String
```
Would be more consistent and expressive

File: runtime/prelude.jakt [hunk 2]
```diff
@@ -31,6 +32,7 @@ extern struct ArraySlice<T> {
...
```
Comment (networkException) [inline runtime/prelude.jakt:35]: Same as above

File: runtime/prelude.jakt (hunk 1 above)
Comment (robryanx) [reply runtime/prelude.jakt:25]: Yeah, if preferable could change this. There is kind of a mix currently in descriptive names and one letter names for anon parameters in prelude currently.
In effect no one would actually use the parameter name in this case.

Comment (networkException) [reply runtime/prelude.jakt:25]: still adding more code that violates the code style also isn't great

Comment (robryanx) [reply runtime/prelude.jakt]: Sure makes sense, I have updated them.

**Review by alimpfard — COMMENTED** [review]
🤔 StringBuilder has a `join` in Serenity, should we do that instead?
```c++
builder.join(Iterable, separator)
```

I'm somewhat reluctant to add overly generic things to the prelude that can be provided by default via (the soon to be ready hopefully) traits

**Comment by robryanx** [comment]
> StringBuilder has a `join` in Serenity, should we do that instead?
> 
> ```c++
> builder.join(Iterable, separator)
> ```
> 
> I'm somewhat reluctant to add overly generic things to the prelude that can be provided by default via (the soon to be ready hopefully) traits

I looked into this first, I think it ends up making the syntax a lot more verbose unless I am missing something
```
let collection = ["A", "B"]
mut sb = StringBuilder::create()
sb.join(",", collection)
    
println("{}", sb.to_string())
```
How would traits interact with the prelude classes? They would just be added on the jakt side?

**Comment by robryanx** [comment]
Will close this and revisit once we can use traits with prelude classes.

**Pull request closed without merging**